mobile/android/logcat-helper.ts

```diff
@@ -18,17 +19,25 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
 		this.mapDevicesLoggingData = Object.create(null);
 	}
 
-	public async start(deviceIdentifier: string): Promise<void> {
+	public async start(options: Mobile.ILogcatStartOptions): Promise<void> {
+		const deviceIdentifier = options.deviceIdentifier;
 		if (deviceIdentifier && !this.mapDevicesLoggingData[deviceIdentifier]) {
-			const adb: Mobile.IDeviceAndroidDebugBridge = this.$injector.resolve(DeviceAndroidDebugBridge, { identifier: deviceIdentifier });
-
-			const logcatStream = await adb.executeCommand(["logcat"], { returnChildProcess: true });
-			const lineStream = byline(logcatStream.stdout);
 			this.mapDevicesLoggingData[deviceIdentifier] = {
-				loggingProcess: logcatStream,
-				lineStream: lineStream
+				loggingProcess: null,
+				lineStream: null,
+				keepSingleProcess: options.keepSingleProcess
 			};
 
+			const adb: Mobile.IDeviceAndroidDebugBridge = this.$injector.resolve(DeviceAndroidDebugBridge, { identifier: deviceIdentifier });
+			const logcatCommand =  ["logcat"];
+			if (options.pid) {
```

Comment:
Just a suggestion. Not a merge stopper.
We can extract a separate method for example in adb service or here in this service that will encapsulate the following logic:
```
const adb: Mobile.IDeviceAndroidDebugBridge = this.$injector.resolve(DeviceAndroidDebugBridge, { identifier: deviceIdentifier });
const logcatCommand =  ["logcat"];
if (options.pid) {
       logcatCommand.push(`--pid=${options.pid}`);
}
const logcatStream = await adb.executeCommand(logcatCommand, { returnChildProcess: true });
return logcatStream;
```